core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/AzureBatchSource.java

```diff
@@ -32,26 +32,28 @@
 @Name("AzureBlobStore")
 @Description("Batch source to read from Azure Blob Storage.")
 public class AzureBatchSource extends FileBatchSource {
-
   @SuppressWarnings("unused")
   private final AzureBatchConfig config;
 
   public AzureBatchSource(AzureBatchConfig config) {
-    super(config);
+    super(new FileBatchConfig(config.referenceName, config.fileRegex, config.timeTable, config.inputFormatClass,
```

Comment:
Since `AzureBatchConfig` and `FileBatchConfig` are type of `FileSourceConfig`, passing the same config throws compilation error. 

![image](https://cloud.githubusercontent.com/assets/14131070/25818565/d332e978-33df-11e7-9b9b-29009622fd66.png)